### PR TITLE
feat: add CraftCommandError

### LINKED ISSUE #260: Add `stderr` to CraftError
### What needs to get done

Store an optional stderr on CraftError, and display it when presenting the error to the user.

### Why it needs to get done

To improve the user's experience when an error on a program that we don't control happens; the clearest example here is during lifecycle builds.

### PR DESCRIPTION
**Pull request by tigarmo**
This new error class has a ``stderr`` attribute that holds the error output of an executed command. As the class' docstring indicates, this new attribute is halfway between ``brief`` and ``details``, verbosity-wise. It's meant to be used in cases where this error output is expected to be useful enough to the end user to make up for the extra text, even in "brief"-mode executions.

Fixes #260

- [ ] Have you followed the guidelines for contributing?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox`?

### CODE REVIEW

File: craft_cli/errors.py [hunk 1]
```diff
@@ -96,6 +96,37 @@ def __eq__(self, other: object) -> bool:
         return NotImplemented
 
 
+class CraftCommandError(CraftError):
+    """A CraftError with precise error output from a command.
+
+    This exception class augments CraftError with the addition of a ``stderr``
+    parameter. This parameter is meant to hold the standard error contents of
+    the failed command. As such, it sits between "brief" and "details" from the
```
Comment (mr-cal) [inline craft_cli/errors.py:104]: `brief` and `details` seem orthogonal - do you mean `brief` and `verbose` or `message` and `details`?

**Review by mr-cal — APPROVED** [review]

File: craft_cli/errors.py (hunk 1 above)
Comment (tigarmo) [reply craft_cli/errors.py:104]: I do mean `message` and `details`, will update. Thankss

**Review by cmatsuoka — APPROVED** [review]
Nice!
